src/BenchmarksServer/Startup.cs

```diff
@@ -424,11 +429,32 @@ private static (string containerId, string imageName) DockerBuildAndRun(string p
             // Only run on the host network on linux
             var useHostNetworking = OperatingSystem == OperatingSystem.Linux;
 
-            var command = useHostNetworking ? $"run -d --rm --network host {imageName}" : 
+            var command = useHostNetworking ? $"run -d --rm --network host {imageName}" :
                                               $"run -d --rm -p {job.Port}:{job.Port} {imageName}";
             var result = ProcessUtil.Run("docker", $"{command} {job.Arguments}");
             var containerId = result.StandardOutput.Trim();
 
+            // Wait until the service is reachable to avoid races where the container started but isn't
+            // listening yet. We only try 5 times, if it keeps failing we ignore it. If the port
+            // is unreachable then clients will fail to connect and the job will be cleaned up properly
+            using (var client = new HttpClient())
```

Comment:
We already have a static `_httpClient` that you can reuse. The server is always on we should reuse the same instance.

Reply:
Not an issue actually, just taste.